lib/appliance_console.rb

```diff
@@ -171,6 +171,10 @@ module ApplianceConsole
 
     clear_screen
 
+    # Calling stty to provide the equivalent line settings when the console is run via an ssh session or
+    # over the virtual machine console.
+    system("stty -echoprt ixany iexten echoe echok")
```

Comment:
It’s in the outer loop to ensure the terminal characteristics are set to a known / working state, whether or not the initial  / default line characteristics cause an issue or the code within the loops (or an external event) changes the characteristics and leaves them set differently.

> On May 13, 2015, at 12:22 PM, Jason Frey <notifications@github.com> wrote:
> 
> In lib/appliance_console.rb <https://github.com/ManageIQ/manageiq/pull/2906#discussion_r30248518>:
> 
> > @@ -171,6 +171,10 @@ module ApplianceConsole
> >  
> >      clear_screen
> >  
> > +    # Calling stty to provide the equivalent line settings when the console is run via an ssh session or
> > +    # over the virtual machine console.
> > +    system("stty -echoprt ixany iexten echoe echok")
> This is being done in the loop...should it be done outside the loop only once?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/ManageIQ/manageiq/pull/2906/files#r30248518>.
>